server/src/main/java/org/elasticsearch/action/admin/cluster/coordination/ClusterFormationInfoAction.java

```diff
@@ -0,0 +1,141 @@
+ * node thinks that cluster formation has failed.
+ */
+public class ClusterFormationInfoAction extends ActionType<ClusterFormationInfoAction.Response> {
+
+    public static final ClusterFormationInfoAction INSTANCE = new ClusterFormationInfoAction();
+    public static final String NAME = "cluster:internal/formation/info";
+
+    private ClusterFormationInfoAction() {
+        super(NAME, ClusterFormationInfoAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+
+        public Request() {}
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            // There are no parameters, so all instances of this class are equal
+            if (this == o) {
+                return true;
+            }
+            return o != null && getClass() == o.getClass();
+        }
+
+        @Override
+        public int hashCode() {
+            // There are no parameters, so all instances of this class are equal
+            return 1;
+        }
+
+    }
+
+    public static class Response extends ActionResponse {
+
+        private final ClusterFormationFailureHelper.ClusterFormationState clusterFormationState;
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            clusterFormationState = new ClusterFormationFailureHelper.ClusterFormationState(in);
+        }
+
+        public Response(ClusterFormationFailureHelper.ClusterFormationState clusterFormationState) {
+            this.clusterFormationState = clusterFormationState;
+        }
+
+        public ClusterFormationFailureHelper.ClusterFormationState getClusterFormationState() {
+            return clusterFormationState;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            clusterFormationState.writeTo(out);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            ClusterFormationInfoAction.Response response = (ClusterFormationInfoAction.Response) o;
+            return clusterFormationState.equals(response.clusterFormationState);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(clusterFormationState);
+        }
+    }
+
+    /**
+     * This transport action fetches the ClusterFormationState from a remote node.
+     */
+    public static class TransportAction extends HandledTransportAction<
```

Comment:
See @DaveCTurner 's comment on the previous PR

https://github.com/elastic/elasticsearch/pull/85782/files#r849933849

Reply:
This is good tho, it's just a simple `HandledTransportAction` here.

Reply:
Ah of course - I just realised it does the needful 
```
        transportService.registerRequestHandler(actionName, executor, false, canTripCircuitBreaker, requestReader, new TransportHandler());
```

Sorry for the noise Keith